x-pack/plugins/task_manager/server/polling/task_poller.ts

```diff
@@ -128,10 +129,12 @@ function asPollingError<T>(err: string | Error, type: PollingErrorType, data: Op
 export class PollingError<T> extends Error {
   public readonly type: PollingErrorType;
   public readonly data: Option<T>;
-  constructor(message: string, type: PollingErrorType, data: Option<T>) {
+  public readonly source: TaskErrorSource;
+  constructor(message: string, type: PollingErrorType, data: Option<T>, source?: TaskErrorSource) {
     super(message);
     Object.setPrototypeOf(this, new.target.prototype);
     this.type = type;
     this.data = data;
+    this.source = source || TaskErrorSource.FRAMEWORK;
```

Comment:
Added the source field to PollingError but as it is always a system error (probably) didn't overwrite the default value in anywhere.

Reply:
so should we just remove the `source?` param, if it wasn't used?  This one does feel like it will always be a framework error ...

Reply:
Re-looking at this, I think we can just set the source to `TaskErrorSource.FRAMEWORK`, and remove the optional `source` option, in the final PR ...